backend/src/events/views.ts

```diff
@@ -29,89 +40,143 @@ export type EventDTO = {
-    key: key,
-    order: order,
-  };
+eventRouter.put(
+  "/:eventid",
+  (authIfAdmin as RequestHandler) || (authIfSupervisor as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    attempt(res, 200, () =>
+      eventController.updateEvent(req.params.eventid, req.body)
+    );
+    socketNotify(`/events/${req.params.eventid}`);
+  }
+);
 
-  const pagination = {
-    after: req.query.after as string,
-    limit: req.query.limit as string,
-  };
+eventRouter.delete(
+  "/:eventid",
+  (authIfAdmin as RequestHandler) || (authIfSupervisor as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    attempt(res, 200, () => eventController.deleteEvent(req.params.eventid));
+    socketNotify("/events");
+  }
+);
+
+eventRouter.get(
+  "/",
+  (authIfAdmin as RequestHandler) || (authIfSupervisor as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    const filter = {
+      date: req.query.date as string,
+      ownerId: req.query.ownerid as string,
+      userId: req.query.userid as string,
+    };
+
+    const sortQuery = req.query.sort as string;
+    const querySplit = sortQuery ? sortQuery.split(":") : ["default", "asc"];
+    const key = querySplit[0];
+    const order = querySplit[1] as Prisma.SortOrder;
+    const sort = {
+      key: key,
+      order: order,
+    };
+
+    const pagination = {
+      after: req.query.after as string,
+      limit: req.query.limit as string,
+    };
+
+    attempt(res, 200, () =>
+      eventController.getEvents(filter, sort, pagination)
+    );
+  }
+);
+
+eventRouter.get(
+  "/upcoming",
+  (authIfAdmin as RequestHandler) ||
+    (authIfSupervisor as RequestHandler) ||
+    (authIfVolunteer as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    attempt(res, 200, eventController.getUpcomingEvents);
+  }
+);
+
+eventRouter.get(
+  "/current",
+  (authIfAdmin as RequestHandler) ||
+    (authIfSupervisor as RequestHandler) ||
+    (authIfVolunteer as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    attempt(res, 200, eventController.getCurrentEvents);
+  }
+);
+
+eventRouter.get(
+  "/past",
+  (authIfAdmin as RequestHandler) ||
+    (authIfSupervisor as RequestHandler) ||
+    (authIfVolunteer as RequestHandler),
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Events']
+    attempt(res, 200, eventController.getPastEvents);
+  }
+);
+
+eventRouter.get(
+  "/:eventid",
+  (authIfAdmin as RequestHandler) ||
+    (authIfSupervisor as RequestHandler) ||
+    (authIfVolunteer as RequestHandler),
+  async (req: Request, res: Response) => {
+    attempt(res, 200, () => eventController.getEvent(req.params.eventid));
+  }
+);
+
+eventRouter.get(
+  "/:eventid/attendees",
+  (authIfAdmin as RequestHandler) || (authIfSupervisor as RequestHandler),
```

Comment:
remember to update this, when you define a variable for both. This applies for ALL places where this is used